src/server.rs

```diff
@@ -131,3 +134,25 @@ async fn shutdown_signal() {
 
     info!("signal received, starting graceful shutdown");
 }
+
+pub struct ServerError(anyhow::Error);
+
+impl IntoResponse for ServerError {
+    fn into_response(self) -> Response {
+        let code = StatusCode::INTERNAL_SERVER_ERROR;
```

Comment:
Isn't this making any error into a `500` class error? 

So if we throw some sort of validation error from `orchestrator.rs` or if downstreams like TGIS throws that error, it would get converted to `500` ?